src/wallet/tx_builder.h

```diff
@@ -145,5 +143,19 @@ cryptonote::transaction finalize_all_proofs_from_transfer_details(
     const fcmp_pp::curve_trees::TreeCacheV1 &tree_cache,
     const fcmp_pp::curve_trees::CurveTreesV1 &curve_trees,
     const cryptonote::account_keys &acc_keys);
+
+wallet2::pending_tx make_pending_carrot_tx(const carrot::CarrotTransactionProposalV1 &tx_proposal,
+    const wallet2::transfer_container &transfers,
+    const cryptonote::account_keys &acc_keys);
```

Comment:
Minor point, but I think it would be nice to make it clear exactly which keys are needed in this function and the `finalize_` functions, rather than pass in `acc_keys`. It helps clarify that the private spend key isn't needed for the former, but needed for the latter, which helps explain these separate functions in the first place.

I also think a comment that explains why they are separate would be nice